docs/source/authentication.rst

```diff
@@ -80,258 +34,83 @@ listed below are configured under the ``auth`` section in the configuration file
-    backend_kwargs = {"file_path": "/path/to/.htpasswd"}
+    backend = pam
     enable = True
-    debug = False
     use_ssl = True
-    cert = /path/to/mycert.crt
-    key = /path/to/mycert.key
-    logging = /path/to/st2auth.logging.conf
-    api_url = http://myhost.example.com:9101/
-
-MongoDB backend
-~~~~~~~~~~~~~~~
-
-Repository URL: https://github.com/StackStorm/st2-auth-backend-mongodb
-
-MongoDB backend supports reading credentials from a MongoDB collection called ``users``. Currently,
-the MongoDB collection and the user entries will have to be generated manually. Entries in this
-``users`` collection need to have the following attributes:
-
-* ``username`` - Username
-* ``salt`` - Password salt
-* ``password`` - SHA256 hash for the salt+password - SHA256(<salt><password>)
-
-**Backend configuration options:**
-
-* ``db_host`` - MongoDB server host.
-* ``db_port`` - MongoDB server port.
-* ``db_name`` - Name of the database to use.
-
-Example ``auth`` section in the |st2| configuration file.
-
-.. sourcecode:: ini
-
-    [auth]
-    mode = standalone
-    backend = mongodb
-    backend_kwargs = {"db_host": "127.0.0.1", "db_port": 27017, "db_name": "st2auth"}
-    enable = True
-    debug = True
-    use_ssl = True
-    cert = /path/to/mycert.crt
-    key = /path/to/mycert.key
-    logging = /path/to/st2auth.logging.conf
-    api_url = http://myhost.example.com:9101/
-
-Keystone backend
-~~~~~~~~~~~~~~~~
-
-Repository URL: https://github.com/StackStorm/st2-auth-backend-keystone
-
-Keystone backend supports authenticating against an OpenStack Keystone auth server.
-
-**Backend configuration options:**
-
-* ``keystone_url`` - Keystone server url, port included (e.g. ``http://keystone.com:5000``).
-* ``keystone_version`` - Keystone APIversion to use. Defaults to ``2``.
-
-Example ``auth`` section in the |st2| configuration file.
-
-.. sourcecode:: ini
-
-    [auth]
-    mode = standalone
-    backend = keystone
-    backend_kwargs = {"keystone_url": "http://keystone.com:5000", "keystone_version": 2}
-    enable = True
-    debug = True
-    use_ssl = True
-    cert = /path/to/mycert.crt
-    key = /path/to/mycert.key
-    logging = /path/to/st2auth.logging.conf
-    api_url = http://myhost.example.com:9101/
-
-LDAP backend
-~~~~~~~~~~~~
-
-LDAP backend reads authentication information from an LDAP server.
-
-Currently there are two types of LDAP backends available - community contributed one and one
-developed and maintained by the StackStorm team. Community contributed one can be installed
-by anyone and the StackStorm developed one is only available in the enterprise edition (for
-more information on the enterprise edition, please see https://stackstorm.com/product/#enterprise).
-
-The difference between them is that the one included in the enterprise edition is developed,
-supported, tested and maintained by the StackStorm team and the community contributed one is
-developed and maintained by the community.
-
-Enterprise LDAP backend
```

Comment:
@m4dcoder I suggest to bring up this section, 
but make it at the same TOC level with  `Authentication Backends` so that user will see it as a second-layer TOC on the left side. And specify inside that "this feature is enterprise only". 

Reason is to drive awareness to enterprise functionality.